lib/adapters/MochaAdapter.js

```diff
@@ -53,60 +52,69 @@ export default class MochaAdapter extends EventEmitter {
       parent = parent.parent
     }
 
-    var test = new Test(
-      testName,
-      suiteName
-    )
-    this.idDataMap[mochaTest.title] = test
+    var test
+
+    if (mochaTest.errors !== undefined) {
+      var status = (mochaTest.state === undefined) ? 'skipped' : mochaTest.state
+
+      test = new Test(testName, suiteName, status, mochaTest.duration,
+          mochaTest.errors)
+    } else {
+      test = new Test(
+        testName,
+        suiteName
+      )
+    }
+
     return test
   }
 
   onStart () {
     var startSuite = this.convertSuite(this.runner.suite)
     startSuite.name = undefined
-    this.idDataMap.globalSuite = startSuite
 
     this.emit('runStart', startSuite)
   }
 
   onSuite (details) {
-    var suite = this.idDataMap[details.title]
-    if (suite !== this.idDataMap.globalSuite) {
-      this.emit('suiteStart', suite)
+    if (!details.root) {
+      this.emit('suiteStart', this.convertSuite(details))
     }
   }
 
   onTest (details) {
     this.errors = []
-    this.emit('testStart', this.idDataMap[details.title])
+    this.emit('testStart', this.convertTest(details))
   }
 
   onFail (test, error) {
     this.errors.push(error)
   }
 
   onTestEnd (details) {
-    var test = this.idDataMap[details.title]
+    var test
+
     if (details.pending) {
+      test = this.convertTest(details)
       this.emit('testStart', test)
-      test.status = 'skipped'
-    } else {
-      test.status = details.state
     }
-    test.runtime = details.duration
-    test.errors = this.errors
+
+    details.errors = this.errors
+    test = this.convertTest(details)
 
     this.emit('testEnd', test)
   }
 
   onSuiteEnd (details) {
-    var suite = this.idDataMap[details.title]
-    if (suite !== this.idDataMap.globalSuite) {
-      this.emit('suiteEnd', suite)
+    if (!details.root) {
+      this.emit('suiteEnd', this.convertSuite(details))
+    }
```

Comment:
Remove linebreak, should be `} else {`. Also verify why linter isn't checking this.